controllers/operatorconfigmap_controller.go

```diff
@@ -291,6 +294,25 @@ func (c *OperatorConfigMapReconciler) Reconcile(ctx context.Context, req ctrl.Re
 				return ctrl.Result{}, err
 			}
 
+			sideCarConfig := &corev1.ConfigMap{}
+			sideCarConfig.Name = csiSidecarConfigMapName
+			sideCarConfig.Namespace = c.OperatorNamespace
+
+			err = c.get(sideCarConfig)
+			if err != nil && !kerrors.IsNotFound(err) {
+				c.log.Error(err, "failed to get csi side car configmap", "name", sideCarConfig)
+				return ctrl.Result{}, err
+			}
+
+			enableOMAPGenerator := false
+			if value, ok := sideCarConfig.Data[csiOMAPGeneratorKey]; ok {
```

Comment:
technically this'll work, however we'll run this code if the CM is not found and so why not check this only if CM is found?